documentation/error-handling.md

```diff
@@ -0,0 +1,113 @@
+# Error handling
+
+Because UI extension scripts run in non-browser environments (Web Worker in web, JavascriptCore on Mobile), it is recommended to manually set up error listeners and error reporting. However, it is possible to use external libraries for error reporting.
+
+## Error handling in UI extensions
+
+UI extensions can throw two types of error: lifecycle errors and non-lifecycle errors.
+
+### Lifecycle error
+
+Most errors will fall into this type. These errors can be thrown when:
+
+- a UI extension script is loaded or imported
+- the UI extension script is run
+- during any callback interaction with UI extension components or APIs
+
+To catch these errors, add an event listener for `unhandledrejection`.
+
+  ```javascript
+  self.onunhandledrejection = (error) => {
+    console.warn('onunhandledrejection', error);
+  };
+  // or
+  self.addEventListener('unhandledrejection', (error) => {
+    console.warn('event unhandledrejection', error);
+  });
+  ```
+
+### Non-lifecycle error
+
+These errors can be thrown when the script encounters an exception while running a `setTimeout` callback. To catch these errors, add an event listener for `error`.
+
+  ```javascript
+  self.onerror = (error) => {
+    console.warn('onerror', error);
+  };
+  // or
+  self.addEventListener('error', (error) => {
+    console.warn('event error', error);
+  });
+  ```
+
+## Handle errors via 3rd-party libraries
+
+### Bugsnag
+
+See https://docs.bugsnag.com/platforms/javascript/.
+
+You can run Bugsnag using the CDN version or npm dependencies. However, Bugsnag requires additional polyfills at runtime, as it will try to access the browser environment. The script below has been tested at the time of writing these docs.
+
+```javascript
+/**
+ * Window polyfill
```

Comment:
Is there an issue on bugsnag for a version that could work in a worker? If not, could we open one? It's very unfortunate that developers need to do this, since they are changing the global environment of a sandbox they are not supposed to control. Maybe we should just add this in our sandbox instead?